python/ray/tune/execution/experiment_state.py

```diff
@@ -380,29 +381,54 @@ def sync_up(self, force: bool = False, wait: bool = False) -> bool:
         self._last_sync_time = now
         return True
 
+    def sync_down_experiment_state(self) -> None:
+        fs = self._storage.storage_filesystem
+        file_infos = fs.get_file_info(
+            pyarrow.fs.FileSelector(self._storage.experiment_fs_path)
+        )
+
+        # Find the latest remote file matching the given pre/suffix in the expdir,
+        # and then copy it down locally for restore.
+        def restore_latest_match(match_prefix: str, match_suffix: str) -> None:
+            matches = []
+            for file_info in file_infos:
+                name = os.path.basename(file_info.path)
+                if name.startswith("basic-variant-state") and name.endswith(".json"):
+                    matches.append(name)
+            if matches:
+                name = max(matches)
+                remote_path = os.path.join(self._storage.experiment_fs_path, name)
+                local_path = os.path.join(self._storage.experiment_local_path, name)
+                logger.info(f"Copying {remote_path} to {local_path}")
+                pyarrow.fs.copy_files(
+                    remote_path,
+                    local_path,
+                    source_filesystem=self._storage.storage_filesystem,
+                )
+            else:
+                logger.info(
+                    "No remote files to restore "
+                    f"matching {match_prefix}.*{match_suffix}"
+                )
+
+        restore_latest_match("basic-variant-state", ".json")
```

Comment:
The searcher state file is a bit more annoying. We only have this `searcher.restore_from_dir` API, so probably we should just download to have minimal API changes.

The filename is not always basic_variant depending on the searcher type.

Glob `self._search_alg.CKPT_FILE_TMPL.format("*")` instead.

And if `_search_alg` is a `SearchGenerator`, you need to download `_search_alg.searcher.CKPT_FILE_TMPL` as well...

Or, maybe let's just download all jsons in there as a hack for 2.7 😄

Reply:
Oh geez.. Yeah maybe let's just download all the *.json *.pkl at the root for now.